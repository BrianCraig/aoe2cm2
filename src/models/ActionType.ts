enum ActionType {
    PICK= "pick",
    BAN = "ban",
    SNIPE = "snipe",
}

export default ActionType;