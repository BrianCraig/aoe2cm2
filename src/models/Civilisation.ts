import GameVersion from "./GameVersion";

enum Name {
    ARABIA = "Arabia",
    ARENA = "Arena",
    CENOTES = "Cenotes",
    GOLDEN_PIT = "Golden Pit",
    HIDEOUT = "Hideout",
    ISLANDS = "Islands",
    MEDITERRANEAN = "Mediterranean",
    SERENGETI = "Serengeti",
    VALLEY = "Valley",
    CROSS = "Cross",
    FORTRESS = "Fortress",
    GRAVEYARDS = "Graveyards",
    LAND_MADNESS = "Land Madness",

    HIDDEN_PICK = "HIDDEN_PICK",
    HIDDEN_BAN = "HIDDEN_BAN",
    HIDDEN_SNIPE = "HIDDEN_SNIPE",
    HIDDEN = "HIDDEN",
    RANDOM = "RANDOM"
}

class Civilisation {

    public static readonly ARABIA:Civilisation = new Civilisation(Name.ARABIA, GameVersion.DEFINITIVE_EDITION);
    public static readonly ARENA:Civilisation = new Civilisation(Name.ARENA, GameVersion.DEFINITIVE_EDITION);
    public static readonly CENOTES:Civilisation = new Civilisation(Name.CENOTES, GameVersion.DEFINITIVE_EDITION);
    public static readonly GOLDEN_PIT:Civilisation = new Civilisation(Name.GOLDEN_PIT, GameVersion.DEFINITIVE_EDITION);
    public static readonly HIDEOUT:Civilisation = new Civilisation(Name.HIDEOUT, GameVersion.DEFINITIVE_EDITION);
    public static readonly ISLANDS:Civilisation = new Civilisation(Name.ISLANDS, GameVersion.DEFINITIVE_EDITION);
    public static readonly MEDITERRANEAN:Civilisation = new Civilisation(Name.MEDITERRANEAN, GameVersion.DEFINITIVE_EDITION);
    public static readonly SERENGETI:Civilisation = new Civilisation(Name.SERENGETI, GameVersion.DEFINITIVE_EDITION);
    public static readonly VALLEY:Civilisation = new Civilisation(Name.VALLEY, GameVersion.DEFINITIVE_EDITION);
    public static readonly CROSS:Civilisation = new Civilisation(Name.CROSS, GameVersion.DEFINITIVE_EDITION);
    public static readonly FORTRESS:Civilisation = new Civilisation(Name.FORTRESS, GameVersion.DEFINITIVE_EDITION);
    public static readonly GRAVEYARDS:Civilisation = new Civilisation(Name.GRAVEYARDS, GameVersion.DEFINITIVE_EDITION);
    public static readonly LAND_MADNESS:Civilisation = new Civilisation(Name.LAND_MADNESS, GameVersion.DEFINITIVE_EDITION);

    public static readonly RANDOM:Civilisation = new Civilisation(Name.RANDOM, GameVersion.DEFINITIVE_EDITION);

    public static readonly HIDDEN_PICK: Civilisation = new Civilisation(Name.HIDDEN_PICK, GameVersion.TECHNICAL);
    public static readonly HIDDEN_BAN: Civilisation = new Civilisation(Name.HIDDEN_BAN, GameVersion.TECHNICAL);
    public static readonly HIDDEN_SNIPE: Civilisation = new Civilisation(Name.HIDDEN_SNIPE, GameVersion.TECHNICAL);
    public static readonly HIDDEN: Civilisation = new Civilisation(Name.HIDDEN, GameVersion.TECHNICAL);

    // DO NOT CHANGE THE ORDER OF ELEMENTS IN THIS ARRAY!!!
    // ONLY APPEND NEW CIVILISATIONS AT THE END!!!
    public static readonly ALL = [
        Civilisation.ARABIA,
        Civilisation.ARENA,
        Civilisation.CENOTES,
        Civilisation.GOLDEN_PIT,
        Civilisation.HIDEOUT,
        Civilisation.ISLANDS,
        Civilisation.MEDITERRANEAN,
        Civilisation.SERENGETI,
        Civilisation.VALLEY,
        Civilisation.CROSS,
        Civilisation.FORTRESS,
        Civilisation.GRAVEYARDS,
        Civilisation.LAND_MADNESS
    ];

    public readonly name:Name;
    public readonly gameVersion:GameVersion;
    public isRandomlyChosenCiv: boolean = false;

    private constructor(name:Name, gameVersion:GameVersion){
        this.name = name;
        this.gameVersion = gameVersion;
    }

}

export default Civilisation;
