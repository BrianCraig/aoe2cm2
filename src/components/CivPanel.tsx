import * as React from 'react';
import ActionType from "../models/ActionType";
import Civilisation from "../models/Civilisation";
import '../pure-min.css'
import '../style2.css'


interface IProps {
    civilisation?: Civilisation;
    active: boolean;
    actionType: ActionType;
}

class CivPanel extends React.Component<IProps, object> {
    public render() {
        const civilisation: Civilisation | undefined = this.props.civilisation;
        let imageSrc: string = "";
        let civilisationName = '';
        if (civilisation !== undefined) {
            imageSrc = "images/civs/" + civilisation.name.toLocaleLowerCase() + "_orig.png";
            civilisationName = civilisation.name;
        }
        let className: string = this.props.actionType.toString() + " card";
        if(this.props.active){
            className += " active-choice";
        }
        let contentClass: string = "box-content";
        if (this.props.civilisation !== undefined) {
            contentClass += " visible";
        }
        return (
            <div className={className}>
                <div className={contentClass}>
                    <div className="stretchy-wrapper">
                        <div className="stretchy-image">
                            <img src={imageSrc}/>
                        </div>
                        <div className="stretchy-text">
                            {civilisationName}
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default CivPanel;
