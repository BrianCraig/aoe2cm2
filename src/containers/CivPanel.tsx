import {IStoreState} from '../types';
import {connect} from 'react-redux';
import CivPanel from "../components/CivPanel";
import ActionType, {actionTypeFromAction} from "../models/ActionType";


export function mapStateToProps(state: IStoreState) {
    let triggerAction: ActionType = ActionType.NOTHING;
    if (state.nextAction < state.preset.turns.length) {
        triggerAction = actionTypeFromAction(state.preset.turns[state.nextAction].action);
    }
    return {
        triggerAction,
        whoAmI: state.whoAmI
    };
}

export function mapDispatchToProps() {
    return {};
}

export default connect(mapStateToProps, mapDispatchToProps)(CivPanel);