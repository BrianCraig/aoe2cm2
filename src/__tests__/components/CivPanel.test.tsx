import {shallow} from "enzyme";
import * as React from 'react';
import CivPanel from "../../components/CivPanel";
import ActionType from "../../models/ActionType";
import Civilisation from "../../models/Civilisation";

it('PICK CivPanels render correctly', () => {
    Civilisation.ALL.forEach((civ: Civilisation) => {
            renderCivPanelAndCheckSnapshot(ActionType.PICK, civ);
        }
    );
});

it('empty PICK CivPanel renders correctly', () => {
    renderCivPanelAndCheckSnapshot(ActionType.PICK);
});

it('BAN CivPanels render correctly', () => {
    Civilisation.ALL.forEach((civ: Civilisation) => {
            renderCivPanelAndCheckSnapshot(ActionType.BAN, civ);
        }
    );
});

it('empty BAN CivPanel renders correctly', () => {
    renderCivPanelAndCheckSnapshot(ActionType.BAN);
});

function renderCivPanelAndCheckSnapshot(actionType: ActionType, civ?: Civilisation) {
    const civpanel = shallow(<CivPanel civilisation={civ} active={false} actionType={actionType}/>);
    expect(civpanel).toMatchSnapshot();
}
