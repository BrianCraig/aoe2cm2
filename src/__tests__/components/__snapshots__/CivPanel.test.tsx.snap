// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BAN CivPanels render correctly 1`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/britons_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Britons
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 2`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/byzantines_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Byzantines
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 3`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/celts_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Celts
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 4`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/chinese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Chinese
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 5`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/franks_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Franks
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 6`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/goths_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Goths
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 7`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/japanese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Japanese
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 8`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/mongols_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Mongols
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 9`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/persians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Persians
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 10`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/saracens_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Saracens
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 11`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/teutons_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Teutons
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 12`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/turks_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Turks
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 13`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/vikings_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Vikings
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 14`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/aztecs_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Aztecs
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 15`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/huns_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Huns
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 16`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/koreans_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Koreans
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 17`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/mayans_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Mayans
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 18`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/spanish_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Spanish
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 19`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/italians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Italians
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 20`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/incas_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Incas
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 21`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/indians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Indians
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 22`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/magyars_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Magyars
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 23`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/slavs_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Slavs
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 24`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/berbers_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Berbers
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 25`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/ethiopians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Ethiopians
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 26`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/malians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Malians
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 27`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/portuguese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Portuguese
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 28`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/burmese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Burmese
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 29`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/khmer_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Khmer
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 30`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/malay_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Malay
      </div>
    </div>
  </div>
</div>
`;

exports[`BAN CivPanels render correctly 31`] = `
<div
  className="ban card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/vietnamese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Vietnamese
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 1`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/britons_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Britons
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 2`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/byzantines_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Byzantines
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 3`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/celts_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Celts
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 4`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/chinese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Chinese
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 5`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/franks_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Franks
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 6`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/goths_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Goths
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 7`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/japanese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Japanese
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 8`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/mongols_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Mongols
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 9`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/persians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Persians
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 10`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/saracens_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Saracens
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 11`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/teutons_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Teutons
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 12`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/turks_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Turks
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 13`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/vikings_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Vikings
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 14`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/aztecs_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Aztecs
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 15`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/huns_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Huns
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 16`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/koreans_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Koreans
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 17`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/mayans_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Mayans
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 18`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/spanish_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Spanish
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 19`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/italians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Italians
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 20`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/incas_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Incas
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 21`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/indians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Indians
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 22`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/magyars_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Magyars
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 23`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/slavs_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Slavs
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 24`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/berbers_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Berbers
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 25`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/ethiopians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Ethiopians
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 26`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/malians_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Malians
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 27`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/portuguese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Portuguese
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 28`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/burmese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Burmese
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 29`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/khmer_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Khmer
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 30`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/malay_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Malay
      </div>
    </div>
  </div>
</div>
`;

exports[`PICK CivPanels render correctly 31`] = `
<div
  className="pick card"
>
  <div
    className="box-content visible"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src="images/civs/vietnamese_orig.png"
        />
      </div>
      <div
        className="stretchy-text"
      >
        Vietnamese
      </div>
    </div>
  </div>
</div>
`;

exports[`empty BAN CivPanel renders correctly 1`] = `
<div
  className="ban card"
>
  <div
    className="box-content"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src=""
        />
      </div>
      <div
        className="stretchy-text"
      />
    </div>
  </div>
</div>
`;

exports[`empty PICK CivPanel renders correctly 1`] = `
<div
  className="pick card"
>
  <div
    className="box-content"
  >
    <div
      className="stretchy-wrapper"
    >
      <div
        className="stretchy-image"
      >
        <img
          src=""
        />
      </div>
      <div
        className="stretchy-text"
      />
    </div>
  </div>
</div>
`;
